brave/brave-core
# chore(wallet): avoid chain/account switches in Send

This PR does the following two refactorings:

1. Send screen no longer relies on `selectedNetwork` and `selectedAccount` backend states.
2. Field information in Send is encoded in the URL query params instead of relying on `React.useState`. This allowed us to cleanup the`useEffect` mess in Send screen.


Resolves https://github.com/brave/brave-browser/issues/34375



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Send flow in desktop and android should continue to work.
2. Must be tested for Ethereum, Solana, Filecoin.
3. Must be tested for ERC20s/SPL and NFTs.
4. Verify whether selecting an asset changes the query params in the URL.
5. Go to **Portfolio** and group entries by **Accounts**. Clicking on Send for an asset should populate it in the Send screen.
6. Open an account under the **Accounts** page. Clicking on Send for an asset should populate it in the Send screen.
